src/classes/CW_BriefApprovalProcessTest.cls

```diff
@@ -0,0 +1,297 @@
+	public static User approverUser;
+	public static User nonApproverUser;
+	public static CW_Campaign__c camp;
+	
+	public static ApproverListSetting__c approverSetting;
+	
+	public static void initUsersAndCustomSettings()
+	{
+    	User thisUser = [SELECT Id FROM User WHERE Id = :UserInfo.getUserId()];
+    	
+		System.runAs(thisUser){
+	    		
+	    	UserRole marketingRole = TestDataUtil.createRole('Marketing Ops');
+	    	
+	    	Group newGroup = TestDataUtil.createGroup('Roadmappers');
+	    	insert newGroup;
+	    	
+	    	approverUser = TestDataUtil.getUserWithProfileName('System Administrator');
+	    	approverUser.UserRoleId = marketingRole.Id;
+	    	approverUser.Department = 'Sport';
+	    	insert approverUser;
+	    	
+	    	nonApproverUser = TestDataUtil.getUserWithProfileName('System Administrator');
+	    	nonApproverUser.UserRoleId = marketingRole.Id;
+	    	nonApproverUser.Department = 'Sport';
+	    	insert nonApproverUser;
+	    	
+	    	approverSetting = new ApproverListSetting__c(Name='1',Username__c=approverUser.Username,Role__c ='Marketing Ops' ,Type__c='Brief');
+	    	insert new List<ApproverListSetting__c>{approverSetting};
+		}
+	}
+	
+	public static void initData()
+	{
+        CW_Brand__c brand = TestDataUtil.createBrand();
+        insert brand;
+        
+        CW_Label__c label = TestDataUtil.createLabel(brand.Id);
+        insert label;
+        
+        CW_Country__c country = TestDataUtil.createCountry();
+        insert country;
+        
+        CW_Product__c product = TestDataUtil.createProduct();
+        insert product;
+        
+        CW_Language__c language = TestDataUtil.createLanguage();
+        insert language;
+        
+        CW_Site_Catalogue__c siteCatalogue = TestDataUtil.createSiteCatalogue(label.Id, country.Id, product.Id);
+        insert siteCatalogue;
+    	
+     	camp = TestDataUtil.createCampaign();
+     	insert camp;
+     	 	
+	    CW_Campaign_Sites__c campSites = TestDataUtil.createCampaignSite(camp.Id, siteCatalogue.Id, language.Id);
+       	insert campSites;
+	    	
+        
+	}
+	
+	public static void updateCampRequiredFields(CW_Campaign__c campaign)
+	{
+        campaign.Suggested_MarCom_Delivery_Date__c = Date.today().addDays(10);
+    	campaign.Main_Campaign_Business_Objective__c = 'test';
+    	campaign.Campaign_Focus__c = 'Retention'; 
+    	campaign.What_is_being_communicated__c ='test';
+    	campaign.Target_Audience_Insight__c = 'test';
+    	campaign.Urgent_Campaign__c = false;
+    	update campaign;
+	    
+	}
+
+    static testMethod void submitBriefForApprovalNoCampaignSite() 
+    {
+    	initUsersAndCustomSettings();
+    	
+    	CW_Campaign__c newCamp = TestDataUtil.createCampaign();
+    	newCamp.OwnerId = nonApproverUser.Id;
+	    insert newCamp;
+	    
+	    updateCampRequiredFields(newCamp);
+    		
+     	System.runAs(nonApproverUser)
+     	{
+     		updateCampRequiredFields(newCamp);	
+	    	
+	        newCamp = [SELECT Name, Campaign_Start_Date__c,
```

Comment:
Problem: Unbound SOQL query. Reason: a SOQL query can return up to 50,000 records. If this number is exceeded a runtime exception is thrown. Suggestion: cap the max number of records returned adding a LIMIT clause to the query.